### DISCUSSION TITLE
Evaluate an exercise that containes main code and functions

### DISCUSSION THREAD

**Discussion by marcothijs** [dodona-edu/feedback-and-support#19] [Questions - Vragen] [answered]
Hi,

Is it possible to use the output and doctest judge in 1 Python excercise. The idee is to have an exercise that ask for code (with input, output, ...) but also request to create specific functions, and I would like to test both. 

I haven't been able to get it work and the documentation seems to suggest it can't be done. 

Thanks for your help.
Marco

**Comment by pdawyndt** [comment, marked as answer]
This is not possible with the Python judge, but can be done using the newer [TESTed](https://docs.dodona.be/en/references/tested/) judge. It has a [domain specific language](https://docs.dodona.be/en/references/tested/dsl/) for specifying tests.
